sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/pathfilters/PathFilterFactory.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.pathfilters
+
+import org.apache.hadoop.conf.Configuration
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.util.CaseInsensitiveMap
+import org.apache.spark.sql.execution.datasources.{FileIndexFilter, ModifiedAfterFilter, ModifiedBeforeFilter, PathGlobFilter}
+
+object PathFilterFactory {
+  PathFilterStrategies.register(ModifiedAfterFilter)
+  PathFilterStrategies.register(ModifiedBeforeFilter)
+  PathFilterStrategies.register(PathGlobFilter)
```

Comment:
We could hard code them into the collection which is aware of path filter strategies but it would be less attractive to do so.  This just provides a very clean interface for accomplishing this objective.